habitat-lab/habitat/config/default_structured_configs.py

```diff
@@ -0,0 +1,1106 @@
+    )
+    base_actions: List[str] = field(
+        default_factory=lambda: ["go_to_relative", "go_to_absolute"]
+    )
+    camera_actions: List[str] = field(
+        default_factory=lambda: ["set_pan", "set_tilt", "set_pan_tilt"]
+    )
+
+
+@dataclass
+class DatasetConfig(HabitatBaseConfig):
+    type: str = "PointNav-v1"
+    split: str = "train"
+    scenes_dir: str = "data/scene_datasets"
+    content_scenes: List[str] = field(default_factory=lambda: ["*"])
+    data_path: str = (
+        "data/datasets/pointnav/"
+        "habitat-test-scenes/v1/{split}/{split}.json.gz"
+    )
+
+
+@dataclass
+class GymConfig(HabitatBaseConfig):
+    auto_name: str = ""
+    obs_keys: Optional[List[str]] = None
+    action_keys: Optional[List[str]] = None
+    achieved_goal_keys: List = field(default_factory=list)
+    desired_goal_keys: List[str] = field(default_factory=list)
+
+
+@dataclass
+class HabitatConfig(HabitatBaseConfig):
+    seed: int = 100
+    # GymHabitatEnv works for all Habitat tasks, including Navigation and
+    # Rearrange. To use a gym environment from the registry, use the
+    # GymRegistryEnv. Any other environment needs to be created and registered.
+    env_task: str = "GymHabitatEnv"
+    # The dependencies for launching the GymRegistryEnv environments.
+    # Modules listed here will be imported prior to making the environment with
+    # gym.make()
+    env_task_gym_dependencies: List = field(default_factory=list)
+    # The key of the gym environment in the registry to use in GymRegistryEnv
+    # for example: `Cartpole-v0`
+    env_task_gym_id: str = ""
+    environment: EnvironmentConfig = EnvironmentConfig()
+    simulator: SimulatorConfig = SimulatorConfig()
+    task: TaskConfig = TaskConfig()
+    dataset: DatasetConfig = DatasetConfig()
+
+
+# -----------------------------------------------------------------------------
+# Register configs in the Hydra ConfigStore
+# -----------------------------------------------------------------------------
+cs = ConfigStore.instance()
+cs.store(group="habitat", name="config", node=HabitatConfig)
+cs.store(
+    group="habitat",
+    name="environment",
+    node=EnvironmentConfig,
+)
+cs.store(
+    group="habitat",
+    name="task",
+    node=TaskConfig,
+)
+cs.store(
+    group="habitat.task.actions.stop",
+    name="stop",
+    node=StopActionConfig,
+)
+cs.store(
+    group="habitat.task.actions.move_forward",
+    name="move_forward",
+    node=MoveForwardActionConfig,
+)
+cs.store(
+    group="habitat.task.actions.turn_left",
+    name="turn_left",
+    node=TurnLeftActionConfig,
+)
+cs.store(
+    group="habitat.task.actions.turn_right",
+    name="turn_right",
+    node=TurnRightActionConfig,
+)
+
+from hydra.core.config_search_path import ConfigSearchPath
+from hydra.core.plugins import Plugins
+from hydra.plugins.search_path_plugin import SearchPathPlugin
+
+
+class DatasetConfigPlugin(SearchPathPlugin):
```

Comment:
Why this class is named `DatasetConfigPlugin`?
Maybe `HabitatConfigPathPlugin`?

Reply:
```suggestion
class HabitatConfigPlugin(SearchPathPlugin):
```